### ISSUE TITLE
Option to interact with nearest object

### ISSUE THREAD

**Issue by TaylanErol** [AngryBeaver/beavers-gamepad#7]
Loved the module, I have tried to used it and found no problems with user movement, but when it comes to openning doors or interacting with stairs I could not find any option to do that. If this were to be implemented this module would be a life saver for in-person dnd sessions.

**Comment by AngryBeaver** [comment]
Yes the opening doors is also on my prio list. Sadly it is more complex or at least the way i wanted it to implement. I am already two month into development of a new module that let you find and execute proximity actions around your token and this will also  come with a submodule inclusion for beavers-gamepad. The latest releases of beavers-gamepad are in preperation for this next module as i need there the context menu and the token rotation.

Currently my first focus had been interaction with wall sections explicitly detecting and revealing secret doors. I have also managed it to locally bridge it to beavers-gamepad as submodule for the ui-component but it is not yet in a state to get released. The next interaction i plan is the open and close door submodule to proximity action as i also need this in my own sessions. 

TLDR:
 I am working on it.
 
I am not sure about the stairs part there are at least 2 modules i know that implement stairs and next there will be foundry v12 that will have something like that as far as i have read. I was under the impression that at least one works with pure movement and as such should have worked when a token enters the section also when moved by gamepads.

What ofc might be a problem is displaying multiple scenes or if you use levels diffrent levels simultaniously for the common display. I have not yet thought about that but yes i think i could write something for it but this has to wait for my other module first.

**Comment by EnoxCode** [comment]
Maybe as an in-between solution I've created some extensions on the great work of @AngryBeaver. Three extensions, but the one most applicable in this issue would be Macros, calling macros with the current controlled token. I used this to trigger doors, end combat turn, turn on light sources, use buttons (with Monk's active tiles), etc. 

[GitHub Repo](https://github.com/EnoxCode/enox-gamepad-extension)

Note: I'm not a programmer, and you need to build and copy the files manually to your modules directory. But hey, it works!

**Comment by AngryBeaver** [comment]
nicely done you even managed to build this while my "how to write extensions" documentation is still missing :-).

this here :  'desc: "beaversGamepad.macro.desc"' can either be a localisation string or a hardcoded description string. When you want to use a localisation string you should namespace it to your package and add a lang direcotry and least one lang.json file into it.

When you managed to get this up and running without beeing programmer thats quite impressive, good job.

DeveloperHint:
npm run devWatch.

With your current copy of the structure you can usually run devWatch to automatically synch changes from your code to your foundry instance.  For this to work your "devDir" in package.json has to pointer to your foundry module directory.
This way you can develop faster without constantly copy to watch changes.



Meanwhile i am currently stucked in my nearsest object detection scripts and ontop do not find much spare time for this project.
